File: app/src/main/java/com/braincorp/githandbook/frontend/ParameterAdapter.java
package com.braincorp.githandbook.frontend;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;
import com.braincorp.githandbook.R;
import com.braincorp.githandbook.backend.BackEndTools;

import java.util.ArrayList;

/**
 * Parameter adapter
 * Created by Alan Camargo - December 2016
 */
public class ParameterAdapter extends ArrayAdapter<String> {

    private Context context;
    private int resource;
    private ArrayList<String> objects;

    /**
     * Creates an instance of ParameterAdapter
     * @param context - the context
     * @param resource - the resource ID
     * @param objects - the objects
     */
    public ParameterAdapter(Context context, int resource, ArrayList<String> objects) {
        super(context, resource, objects);
        this.context = context;
        this.resource = resource;
        this.objects = objects;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ParameterHolder holder = new ParameterHolder();
        View row = convertView;
        if (row == null) {
            LayoutInflater inflater = ((Activity)context).getLayoutInflater();
            row = inflater.inflate(resource, parent, false);
            holder.parameterRow = (TextView)row.findViewById(R.id.paramRow);
            row.setTag(holder);
        } else
            holder = (ParameterHolder)row.getTag();
        String param = objects.get(position);
        if (param == null)
            param = "<NULL>";
        else
            param = context.getString(BackEndTools.getStringResourceKey(context, param));
        holder.parameterRow.setText(param);
        return row;
    }

    private static class ParameterHolder {

        TextView parameterRow;

    }

}

File: app/src/main/kotlin/com/braincorp/githandbook/commands/ui/adapter/ParameterAdapter.kt
package com.braincorp.githandbook.commands.ui.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.braincorp.githandbook.commands.ui.model.UiParameter
import com.braincorp.githandbook.databinding.ItemParameterBinding

class ParameterAdapter(
    private val onItemClicked: (UiParameter) -> Unit
) : RecyclerView.Adapter<ParameterViewHolder>() {

    private var parameters: List<UiParameter> = emptyList()

    fun submitList(parameters: List<UiParameter>) {
        this.parameters = parameters
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ParameterViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val binding = ItemParameterBinding.inflate(inflater, parent, false)
        return ParameterViewHolder(binding, onItemClicked)
    }

    override fun onBindViewHolder(holder: ParameterViewHolder, position: Int) {
        val command = parameters[position]
        holder.bindTo(command)
    }

    override fun getItemCount() = parameters.size
}
